domain/src/main/kotlin/no/nav/su/se/bakover/domain/søknadsbehandling/Søknadsbehandling.kt

```diff
@@ -1345,47 +1345,40 @@ sealed class KunneIkkeIverksette {
     object KunneIkkeGenerereVedtaksbrev : KunneIkkeIverksette()
 }
 
-private fun List<Søknadsbehandling>.kastHvisDetFinnesEnAnnenÅpenBehandling(behandlingId: UUID) {
-    this
-        .filter { b -> b.id != behandlingId }
-        .filterNot { it.status == BehandlingsStatus.IVERKSATT_INNVILGET || it.status == BehandlingsStatus.IVERKSATT_AVSLAG }
-        .ifNotEmpty {
-            throw IllegalStateException("Det finnes flere behandlinger under behandling")
-        }
-    // TODO jah: Kan på sikt flytte disse valideringene til Sak og/eller database-laget.
-}
-
-private fun List<Søknadsbehandling>.kastHvisDetFinnesEnÅpenBehandling() {
-    this
-        .filterNot { it.status == BehandlingsStatus.IVERKSATT_INNVILGET || it.status == BehandlingsStatus.IVERKSATT_AVSLAG }
-        .ifNotEmpty {
-            throw IllegalStateException("Det finnes flere behandlinger under behandling")
-        }
-    // TODO jah: Kan på sikt flytte disse valideringene til Sak og/eller database-laget.
-}
-
 fun List<Søknadsbehandling>.hentSøknadstypeUtenBehandling(): Søknadstype {
-    this.kastHvisDetFinnesEnÅpenBehandling()
-
-    this
-        .filter { it.status == BehandlingsStatus.IVERKSATT_INNVILGET }
-        .ifNotEmpty {
-            return Søknadstype.NY_PERIODE
-        }
-
+    if (any { it.status == BehandlingsStatus.IVERKSATT_INNVILGET }) {
+        return Søknadstype.NY_PERIODE
+    }
+    if (this.filterNot { it.status == BehandlingsStatus.IVERKSATT_AVSLAG }.isNotEmpty()) {
+        // TODO jah: Dette vil stopp nye søknadsinnsendinger dersom saken allerede har en åpen søknadsbehandling.
```

Comment:
Hensiktsmessig?

Reply:
Dette er jo en fagavklaring som gjenstår. Men det feiler for Maria i test nå og dette vil gjøre at det feiler mindre. Dersom vi ikke ønsker at det skal feile må vi separere det å sende inn søknad og det å lage oppgaver.